src/compiler/typescript_to_rust/azle_generate/src/canister_method/annotated_fn_decl/mod.rs

```diff
@@ -153,47 +151,38 @@ impl SourceMapped<'_, AnnotatedFnDecl> {
         self.annotation.method_type == canister_method_type
     }
 
-    pub fn is_manual(&self) -> Result<bool, Error> {
+    pub fn is_manual(&self) -> bool {
         let return_type = match &self.fn_decl.function.return_type {
-            Some(ts_type_ann) => match self.is_promise()? {
+            Some(ts_type_ann) => match self.is_promise() {
                 true => match &ts_type_ann.type_ann.as_ts_type_ref() {
                     Some(ts_type_ref) => match &ts_type_ref.type_params {
                         Some(type_param_instantiation) => &type_param_instantiation.params[0],
-                        None => return Ok(false),
+                        None => return false,
                     },
-                    None => return Ok(false),
+                    None => return false,
                 },
                 false => &*ts_type_ann.type_ann,
             },
-            None => return Ok(false),
+            None => return false,
```

Comment:
This seems like this `let return_type = ...` block could be refined. Either question mark syntax or something. SO... just explore how to make this better.

Reply:
I'm not convinced it can be refined. Lets talk about it. Here is what I have so far, in the mean time I rolling back the changes
```rust
    pub fn is_manual(&self) -> bool {
        let return_type = match &self.fn_decl.function.return_type {
            Some(ts_type_ann) => get_promised_type(ts_type_ann),
            None => None,
        };

        let return_type = match return_type {
            Some(return_type) => return_type,
            None => return false,
        };

        match return_type {
            TsType::TsTypeRef(ts_type_ref) => self
                .symbol_table
                .manual
                .contains(&self.spawn(ts_type_ref).get_name()),

            _ => false,
        }
    }

    pub fn is_promise(&self) -> bool {
        match &self.fn_decl.function.return_type {
            Some(ts_type_ann) => is_promise(ts_type_ann),
            None => false,
        }
    }
}

fn is_promise(ts_type_ann: &TsTypeAnn) -> bool {
    match &*ts_type_ann.type_ann {
        TsType::TsTypeRef(ts_type_ref) => ts_type_ref.type_name.get_name() == "Promise",
        _ => false,
    }
}

// Get's the Ts Type of a type annotation. If it's a promise it returns the promised type
// This is no good. there are a lot of implications to this function that I don't like
// For starters. what are we doing getting the promised type here... especially when it's not a promise.
// But that's not too bad, we're just getting a name...
// What the real problem is that before it used to be "is it malformed? Don't
// care, we just wont consider it," This version will just give you none and no
// indication if it was malformed. That's fine if you only use it for the one
// time that we use But if you look at this name and you're like, oh great I can
// use this anytime I want to get the inside out then you are going to be
// disappointed by it's lack of error handling
fn get_promised_type(ts_type_ann: &TsTypeAnn) -> Option<&TsType> {
    if is_promise(ts_type_ann) {
        if let Some(ts_type_ref) = ts_type_ann.type_ann.as_ts_type_ref() {
            if let Some(type_param_instantiation) = &ts_type_ref.type_params {
                return Some(&type_param_instantiation.params[0].as_ref());
            }
        }
    } else {
        return Some(&*ts_type_ann.type_ann);
    }
    None
}
```

Reply:
We got it working